### PR TITLE
test(api): add structure to test api endpoints

### PR DESCRIPTION
**Pull request by kduret**
## Description

* add structure to test api endpoints
* test host monitoring listing endpoint

**Fixes** Refs: MON-5020

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 19.04.x
- [ ] 19.10.x
- [ ] 20.04.x
- [x] 20.10.x (master)

<h2> How this pull request can be tested ? </h2>

Check CI

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -26,11 +26,12 @@
         "adlawson/vfs": "^0.12.1",
         "zircote/swagger-php": "^3.0",
         "symfony/phpunit-bridge": "4.4.*",
-        "centreon/centreon-test-lib": "dev-master",
+        "centreon/centreon-test-lib": "dev-MON-5020-api-test",
```
Comment (sc979) [inline composer.json:29]: shouldn't you revert it to master, as your tests succeeded ?

Comment (kduret) [reply composer.json:29]: I wait to improve centreon-test-lib branch. Once it's done and merge, I'll put back dev-master

**Comment by sc979** [comment]
@ganoze , it' not intended for unit test, nor end-to-end tests, so a different step is a goog idea.
These test are related to the API v2, so if they fail, they'll impact different features. This step needs to pass before executing the old ones.
And as the behat's end-to-end tests may be replaced by circle-ci. Distinguish them from the new API ones is consistent

**Comment by kduret** [comment]
> Why is this set as a new stage ? Could it be set as a classical acceptance test or do we want to deliberately check API before other acceptance tests and hence increase the total pipeline time ? Are the API tests expected to detect a broad range of errors ?

@ganoze , The purpose is to separate e2e tests from integration tests.
e2e depends from API integration tests, so we should test these before to avoid unnecessary tests

File: tests/api/features/ServiceMonitoring.feature [hunk 2]
```diff
@@ -0,0 +1,33 @@
+Feature:
+    In order to monitor services
+    As a user
+    I want to get service information using api
+
+    Background:
+        Given a running instance of Centreon Web API
+
+    Scenario: Browse Services
```
Comment (sc979) [inline tests/api/features/ServiceMonitoring.feature:9]: As details and list are related to browsing, could you change the scenario title ?

**Comment by sc979** [comment]
@kduret don't forget to change the reference branch before merging it ;)

**Pull request merged**